Dockerfile

```diff
@@ -6,16 +6,18 @@ RUN apt-get update && \
 RUN curl -L -o /usr/bin/solc https://github.com/ethereum/solidity/releases/download/v0.4.21/solc-static-linux && \
     chmod +x /usr/bin/solc
 
+COPY ./constraints.txt /relay/constraints.txt
 COPY ./requirements.txt /relay/requirements.txt
 
 WORKDIR /relay
 
-RUN pip install -r requirements.txt
+RUN pip install -c constraints.txt populus
```

Comment:
Why is this extra install step necessary?

Reply:
If I install the thing into a fresh virtualenv, it complains while building trustlines-contracts that populus is missing. I guess this could be resolved by putting a proper release on pypi, which bundles the compiled contracts.

```
...
  Running setup.py bdist_wheel for trustlines-contracts ... error
  Complete output from command /home/ralf/tt/bin/python3 -u -c "import setuptools, tokenize;__file__='/tmp/pip-req-build-94xhpk_2/setup.py';f=getattr(tokenize, 'open', open)(__file__);code=f.read().replace('\r\n', '\n');f.close();exec(compile(code, __file__, 'exec'))" bdist_wheel -d /tmp/pip-wheel-lkwjhgdb --python-tag cp36:
  running bdist_wheel
  running build
  running build_py
  creating build
  creating build/lib
  creating build/lib/tlcontracts
  copying tlcontracts/signing.py -> build/lib/tlcontracts
  copying tlcontracts/deploy.py -> build/lib/tlcontracts
  copying tlcontracts/__init__.py -> build/lib/tlcontracts
  copying tlcontracts/exchange.py -> build/lib/tlcontracts
  running compile_contracts
  Traceback (most recent call last):
    File "<string>", line 1, in <module>
    File "/tmp/pip-req-build-94xhpk_2/setup.py", line 149, in <module>
      'build_py': BuildPyCommand,
    File "/home/ralf/tt/lib/python3.6/site-packages/setuptools/__init__.py", line 129, in setup
      return distutils.core.setup(**attrs)
    File "/usr/local/stow/python-3.6.5/lib/python3.6/distutils/core.py", line 148, in setup
      dist.run_commands()
    File "/usr/local/stow/python-3.6.5/lib/python3.6/distutils/dist.py", line 955, in run_commands
      self.run_command(cmd)
    File "/usr/local/stow/python-3.6.5/lib/python3.6/distutils/dist.py", line 974, in run_command
      cmd_obj.run()
    File "/home/ralf/tt/lib/python3.6/site-packages/wheel/bdist_wheel.py", line 202, in run
      self.run_command('build')
    File "/usr/local/stow/python-3.6.5/lib/python3.6/distutils/cmd.py", line 313, in run_command
      self.distribution.run_command(command)
    File "/usr/local/stow/python-3.6.5/lib/python3.6/distutils/dist.py", line 974, in run_command
      cmd_obj.run()
    File "/usr/local/stow/python-3.6.5/lib/python3.6/distutils/command/build.py", line 135, in run
      self.run_command(cmd_name)
    File "/usr/local/stow/python-3.6.5/lib/python3.6/distutils/cmd.py", line 313, in run_command
      self.distribution.run_command(command)
    File "/usr/local/stow/python-3.6.5/lib/python3.6/distutils/dist.py", line 974, in run_command
      cmd_obj.run()
    File "/tmp/pip-req-build-94xhpk_2/setup.py", line 31, in run
      self.run_command('compile_contracts')
    File "/usr/local/stow/python-3.6.5/lib/python3.6/distutils/cmd.py", line 313, in run_command
      self.distribution.run_command(command)
    File "/usr/local/stow/python-3.6.5/lib/python3.6/distutils/dist.py", line 974, in run_command
      cmd_obj.run()
    File "/tmp/pip-req-build-94xhpk_2/setup.py", line 45, in run
      from populus import Project
  ModuleNotFoundError: No module named 'populus'
  
  ----------------------------------------
  Failed building wheel for trustlines-contracts
  Running setup.py clean for trustlines-contracts
Successfully built cytoolz gevent geventhttpclient networkx pygraphviz SQLAlchemy
Failed to build trustlines-contracts
Installing collected packages: aniso8601, anyconfig, argh, bitcoin, certifi, pycparser, cffi, chardet, click, contextlib2, coverage, toolz, cytoolz, decorator, pysha3, eth-utils, eth-abi, eth-keys, semantic-version, rlp, eth-tester, Werkzeug, json-rpc, secp256k1, repoze.lru, scrypt, pyethash, PyYAML, pycryptodome, pbkdf2, ethereum, eth-testrpc, pyflakes, pycodestyle, mccabe, flake8, MarkupSafe, Jinja2, itsdangerous, Flask, six, Flask-Cors, pytz, Flask-RESTful, greenlet, gevent, gevent-websocket, Flask-Sockets, geventhttpclient, idna, jsonschema, lru-dict, marshmallow, typed-ast, mypy, networkx, pathtools, toposort, py-geth, py-solc, pylru, py, pytest, urllib3, requests, web3, watchdog, populus, pygraphviz, SQLAlchemy, tinyrpc, webargs, trustlines-contracts
  Running setup.py install for trustlines-contracts ... done
Successfully installed Flask-0.12.2 Flask-Cors-3.0.3 Flask-RESTful-0.3.6 Flask-Sockets-0.2.1 Jinja2-2.10 MarkupSafe-1.0 PyYAML-3.12 SQLAlchemy-1.2.7 Werkzeug-0.14.1 aniso8601-3.0.0 anyconfig-0.9.4 argh-0.26.2 bitcoin-1.1.42 certifi-2018.4.16 cffi-1.11.5 chardet-3.0.4 click-6.7 contextlib2-0.5.5 coverage-4.5.1 cytoolz-0.9.0.1 decorator-4.3.0 eth-abi-0.5.0 eth-keys-0.1.0b4 eth-tester-0.1.0b11 eth-testrpc-1.3.3 eth-utils-0.7.4 ethereum-1.6.1 flake8-3.5.0 gevent-1.1.2 gevent-websocket-0.10.1 geventhttpclient-1.3.1 greenlet-0.4.13 idna-2.6 itsdangerous-0.24 json-rpc-1.10.8 jsonschema-2.6.0 lru-dict-1.1.6 marshmallow-2.15.0 mccabe-0.6.1 mypy-0.590 networkx-2.1 pathtools-0.1.2 pbkdf2-1.3 populus-2.2.0 py-1.5.3 py-geth-2.0.1 py-solc-2.1.0 pycodestyle-2.3.1 pycparser-2.18 pycryptodome-3.6.1 pyethash-0.1.27 pyflakes-1.6.0 pygraphviz-1.3.1 pylru-1.0.9 pysha3-1.0.2 pytest-3.2.5 pytz-2018.4 repoze.lru-0.7 requests-2.18.4 rlp-0.6.0 scrypt-0.8.6 secp256k1-0.13.2 semantic-version-2.6.0 six-1.11.0 tinyrpc-0.8 toolz-0.9.0 toposort-1.5 trustlines-contracts-0.0.1.dev1 typed-ast-1.1.0 urllib3-1.22 watchdog-0.8.3 web3-3.16.5 webargs-2.1.0
```

Reply:
Mmh but https://github.com/trustlines-network/contracts/blob/develop/setup.py#L116 declares it as a dependency so it should be installed right? Mmh we never had to do that manual step before, and I dont like it, because populus is not a direct dependency for the relay server. We should try to fix that.

Reply:
But it may not be available when building. Maybe setting setup_requires=... helps.

Reply:
Ah I actually saw this error on Come's computer too, but even tho there was this error, the contracts were build. Mmh.